libs/utils/ctx.go

```diff
@@ -1,24 +1,22 @@
-package getters
+package utils
 
 import (
 	"context"
-	"errors"
 	"time"
-
-	logging "github.com/ipfs/go-log/v2"
-	"go.opentelemetry.io/otel"
 )
 
-var (
-	tracer = otel.Tracer("share/getters")
-	log    = logging.Logger("share/getters")
+// ResetContextOnError returns a fresh context if the given context has an error.
+func ResetContextOnError(ctx context.Context) context.Context {
+	if ctx.Err() != nil {
+		ctx = context.Background()
```

Comment:
I know that this is an old func, but should we change to something 'safer' ? `context.WithTimeout(time.Second)` by example. WDYT ?

Reply:
I think it makes sense, but not in this PR. Perhaps time.Second should be passed to function in. Like
```
NewContextOnContextErr(ctx, newTimeout time.Time)
```